src/main/java/com/yelp/nrtsearch/server/plugins/AnalysisPlugin.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.yelp.nrtsearch.server.plugins;
+
+import com.yelp.nrtsearch.server.luceneserver.analysis.AnalysisProvider;
+
+import org.apache.lucene.analysis.Analyzer;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Plugin interface for providing custom analysis implementations. Currently allows for the registration
+ * of named {@link Analyzer} providers. These analyzers will be used when the name is provides as the
+ * predefined AnalyzerType for queries and field registration.
+ */
+public interface AnalysisPlugin {
```

Comment:
If there are any issues with the current custom analyzer api, why not extend the current api instead of adding analysis plugin ?

Reply:
Extending the custom analyzer api would limit you to just functionality provided by lucene/nrtsearch. This interface allows you to provide a class that extends the lucene Analyzer and does arbitrary analysis processing.

Reply:
+1 Makes sense